sklearn/preprocessing/tests/test_data.py

```diff
@@ -1651,6 +1651,17 @@ def test_robust_scaler_zero_variance_features():
     assert_array_almost_equal(X_trans_new, X_expected_new, decimal=3)
 
 
+def test_robust_scaler_unit_variance():
+    # Check RobustScaler with unit_variance=True is equivalent to
+    # StandardScaler on large normal data
+    rng = np.random.RandomState(0)
+    X = rng.randn(100000, 1) * 5 + 2
+    robust_scaler = RobustScaler(unit_variance=True)
+    standard_scaler = StandardScaler()
+    assert_allclose(robust_scaler.fit_transform(X),
```

Comment:
Actually, the test is wrong. `StandardScaler` will center considering the mean, while the `RobustScaler` will center considering the median. So we will always have too much changes.

I would do the following test then:

```python
rng = np.random.RandomState(42)
X = rng.randn(1000000, 1)
X_with_outliers = np.vstack(
    [X, np.ones((100, 1)) * 100, np.ones((100, 1)) * -100]
)

quantile_range = (1, 99)
robust_scaler = RobustScaler(
    quantile_range=quantile_range, unit_variance=True
).fit(X_with_outliers)
X_trans = robust_scaler.transform(X)


assert robust_scaler.center_ == pytest.approx(0, abs=1e-3)
assert robust_scaler.scale_ == pytest.approx(1, abs=1e-2)
q_min_emperical, q_max_emperical = np.percentile(X, quantile_range)
assert robust_scaler.adjust_ == pytest.approx(
    q_max_emperical - q_min_emperical, abs=1e-2
)
assert X_trans.std() == pytest.approx(1, abs=1e-2)
```

Reply:
In this case, we check that the statistics computed would be equivalent to a test set without outliers. We need some absolute error larger since we compute empirical stats.

Reply:
We would also need to add some comments :)

@lucyleeow let me know if it makes sense and if I did not misunderstood something.

Reply:
Looks good to me, it's a better test.
I guess the assumption was that mean ~= median for normal data but I think the sample size req for this is too large.